crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -600,31 +600,33 @@ impl TimelineStateTransaction<'_> {
                             .map_or(RemoteEventOrigin::Unknown, |item| item.origin),
                     };
 
-                    match origin {
-                        RemoteEventOrigin::Sync | RemoteEventOrigin::Unknown => {
-                            should_add = match self.timeline_focus {
-                                TimelineFocusKind::PinnedEvents => {
-                                    // Only insert timeline items for pinned events, if the event
-                                    // came from the sync.
-                                    room_data_provider.is_pinned_event(&event_id)
-                                }
-
-                                TimelineFocusKind::Live => {
-                                    // Always add new items to a live timeline receiving items from
-                                    // sync.
-                                    true
-                                }
-
-                                TimelineFocusKind::Event => {
-                                    // Never add any item to a focused timeline when the item comes
-                                    // down from the sync.
-                                    false
-                                }
-                            };
+                    // If the event should be added according to the general event filter, use a
+                    // second filter to decide whether it should be added depending on the timeline
+                    // focus and events origin, if needed
+                    match self.timeline_focus {
+                        TimelineFocusKind::PinnedEvents => {
+                            // Only add pinned events for the pinned events timeline
+                            should_add = room_data_provider.is_pinned_event(&event_id);
                         }
-
-                        RemoteEventOrigin::Pagination | RemoteEventOrigin::Cache => {
-                            // Forward the previous decision to add it.
+                        TimelineFocusKind::Live => {
+                            if matches!(
```

Comment:
Here and below, we could use an explicit `match` statement; this way, if we added `RemoteEventOrigin` variants in the future, it would have us pause and think what we need to do here.